ext/utils/queue.h

```diff
@@ -0,0 +1,522 @@
+/*
```

Comment:
Have you considered using `<misc/dlist.h>` instead?

Reply:
i looked into dlist.h - I will try to use it instead queue.h